---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, include = FALSE}
knitr::opts_chunk$set(
    collapse = TRUE,
    comment = "#>",
    fig.path = "man/figures/README-",
    out.width = "100%"
)
```

# cageminer <img src='man/figures/logo.png' align="right" height="139" />

<!-- badges: start -->
[![GitHub issues](https://img.shields.io/github/issues/almeidasilvaf/cageminer)](https://github.com/almeidasilvaf/cageminer/issues)
[![Lifecycle: stable](https://img.shields.io/badge/lifecycle-stable-brightgreen.svg)](https://lifecycle.r-lib.org/articles/stages.html#stable)
[![R-CMD-check-bioc](https://github.com/almeidasilvaf/cageminer/workflows/R-CMD-check-bioc/badge.svg)](https://github.com/almeidasilvaf/cageminer/actions)
[![Codecov test coverage](https://codecov.io/gh/almeidasilvaf/cageminer/branch/master/graph/badge.svg)](https://codecov.io/gh/almeidasilvaf/cageminer?branch=master)
<!-- badges: end -->

The goal of `cageminer` is to integrate SNP data from GWAS results with gene coexpression networks to identify high-confidence candidate genes involved in a particular phenotype. To identify high-confidence candidate genes, `cageminer`
considers 3 criteria:

1. Physical proximity (or linkage disequilibrium with) trait-related SNPs;
2. Presence in coexpression modules enriched in guide genes (i.e.,
"reference" genes that are known to be associated with the phenotype).
3. Significant altered expression levels in a condition of interest (e.g., 
stress, disease, etc).

By default, `cageminer` defines genes as high-confidence candidates if they
satisfy all of the 3 criteria above, but users can choose to use only one/some
of them.

## Installation instructions

Get the latest stable `R` release from [CRAN](http://cran.r-project.org/). Then install `cageminer` from [Bioconductor](http://bioconductor.org/) using the following code:

```{r 'install', eval = FALSE}
if (!requireNamespace("BiocManager", quietly = TRUE)) {
    install.packages("BiocManager")
}

BiocManager::install("cageminer")
```

And the development version from [GitHub](https://github.com/almeidasilvaf/cageminer) with:

```{r 'install_dev', eval = FALSE}
BiocManager::install("almeidasilvaf/cageminer")
```

## Citation

Below is the citation output from using `citation('cageminer')` in R. Please
run this yourself to check for any updates on how to cite __cageminer__.

```{r 'citation', eval = requireNamespace('cageminer')}
print(citation('cageminer'), bibtex = TRUE)
```

## Code of Conduct

Please note that the `cageminer` project is released with a [Contributor Code of Conduct](http://bioconductor.org/about/code-of-conduct/). By contributing to this project, you agree to abide by its terms.

## Development tools

* Continuous code testing is possible thanks to [GitHub actions](https://www.tidyverse.org/blog/2020/04/usethis-1-6-0/)  through `r BiocStyle::CRANpkg('usethis')`, `r BiocStyle::CRANpkg('remotes')`, and `r BiocStyle::CRANpkg('rcmdcheck')` customized to use [Bioconductor's docker containers](https://www.bioconductor.org/help/docker/) and `r BiocStyle::Biocpkg('BiocCheck')`.
* Code coverage assessment is possible thanks to [codecov](https://codecov.io/gh) and `r BiocStyle::CRANpkg('covr')`.
* The [documentation website](http://almeidasilvaf.github.io/cageminer) is automatically updated thanks to `r BiocStyle::CRANpkg('pkgdown')`.
* The code is styled automatically thanks to `r BiocStyle::CRANpkg('styler')`.
* The documentation is formatted thanks to `r BiocStyle::CRANpkg('devtools')` and `r BiocStyle::CRANpkg('roxygen2')`.

For more details, check the `dev` directory.

This package was developed using `r BiocStyle::Biocpkg('biocthis')`.


