<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Mine high-confidence candidate genes in a single step — mine_candidates • cageminer</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Mine high-confidence candidate genes in a single step — mine_candidates"><meta property="og:description" content="Mine high-confidence candidate genes in a single step"><meta property="og:image" content="/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">cageminer</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.3.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/cageminer.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/almeidasilvaf/cageminer/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Mine high-confidence candidate genes in a single step</h1>
    <small class="dont-index">Source: <a href="https://github.com/almeidasilvaf/cageminer/blob/HEAD/R/candidate_mining.R" class="external-link"><code>R/candidate_mining.R</code></a></small>
    <div class="hidden name"><code>mine_candidates.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Mine high-confidence candidate genes in a single step</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">mine_candidates</span><span class="op">(</span></span>
<span>  gene_ranges <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  marker_ranges <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  window <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  expand_intervals <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  gene_col <span class="op">=</span> <span class="st">"ID"</span>,</span>
<span>  exp <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  gcn <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  guides <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  <span class="va">metadata</span>,</span>
<span>  <span class="va">sample_group</span>,</span>
<span>  min_cor <span class="op">=</span> <span class="fl">0.2</span>,</span>
<span>  alpha <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  continuous <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>gene_ranges</dt>
<dd><p>A GRanges object with genomic coordinates
of all genes in the genome.</p></dd>


<dt>marker_ranges</dt>
<dd><p>Genomic positions of SNPs. For a single trait,
a GRanges object. For multiple traits, a GRangesList or CompressedGRangesList
object, with each element of the list representing SNP positions for a
particular trait.</p></dd>


<dt>window</dt>
<dd><p>Sliding window (in Mb) upstream and downstream relative
to each SNP. Default: 2.</p></dd>


<dt>expand_intervals</dt>
<dd><p>Logical indicating whether or not to expand markers
that are represented by intervals. This is particularly useful
if users want to use a custom interval defined by linkage disequilibrium,
for example. Default: TRUE.</p></dd>


<dt>gene_col</dt>
<dd><p>Column of the GRanges object containing gene ID.
Default: "ID", the default for gff/gff3 files imported with
rtracklayer::import.</p></dd>


<dt>exp</dt>
<dd><p>Expression data frame with genes in row names and samples in
column names or a SummarizedExperiment object.</p></dd>


<dt>gcn</dt>
<dd><p>Gene coexpression network returned by <code><a href="https://rdrr.io/pkg/BioNERO/man/exp2gcn.html" class="external-link">BioNERO::exp2gcn()</a></code>.</p></dd>


<dt>guides</dt>
<dd><p>Guide genes as a character vector or as a data frame with
genes in the first column and gene annotation class in the second column.</p></dd>


<dt>metadata</dt>
<dd><p>Sample metadata with samples in row names and sample
information in the first column. Ignored if <code>exp</code> is a SummarizedExperiment
object, as the colData will be extracted from the object.</p></dd>


<dt>sample_group</dt>
<dd><p>Level of sample metadata to be used for filtering
in gene-trait correlation.</p></dd>


<dt>min_cor</dt>
<dd><p>Minimum correlation value for
<code><a href="https://rdrr.io/pkg/BioNERO/man/gene_significance.html" class="external-link">BioNERO::gene_significance()</a></code>. Default: 0.2</p></dd>


<dt>alpha</dt>
<dd><p>Numeric indicating significance level. Default: 0.05</p></dd>


<dt>continuous</dt>
<dd><p>Logical indicating whether metadata is continuous or not.
Default: FALSE</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>A data frame with mined candidate genes and their correlation to
the condition of interest.</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co"># \donttest{</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">pepper_se</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">snp_pos</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">gene_ranges</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">guides</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">gcn</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">candidates</span> <span class="op">&lt;-</span> <span class="fu">mine_candidates</span><span class="op">(</span><span class="va">gene_ranges</span>, <span class="va">snp_pos</span>, exp <span class="op">=</span> <span class="va">pepper_se</span>,</span></span>
<span class="r-in"><span>                              gcn <span class="op">=</span> <span class="va">gcn</span>, guides <span class="op">=</span> <span class="va">guides</span>,</span></span>
<span class="r-in"><span>                              sample_group <span class="op">=</span> <span class="st">"PRR_stress"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Enrichment analysis for module cyan...</span>
<span class="r-plt img"><img src="mine_candidates-1.png" alt="" width="700" height="433"></span>
<span class="r-in"><span><span class="co"># }</span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Fabrício Almeida-Silva, Thiago Venancio.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.5.</p>
</div>

      </footer></div>

  


  

  </body></html>

